affectedmoduledetector/src/main/kotlin/com/dropbox/affectedmoduledetector/AffectedModuleDetector.kt

```diff
@@ -366,7 +366,9 @@ class AffectedModuleDetectorImpl constructor(
         val isRootProject = project.isRoot
         val isProjectAffected = affectedProjects.contains(project)
         val isProjectProvided = isProjectProvided2(project)
-        val isNotModuleExcluded = !config.excludedModules.contains(project.name)
+        val isModuleExcludedByName = config.excludedModules.contains(project.name)
+        val isModuleExcludedByRegex = config.excludedModules.any { project.path.matches(it.toRegex()) }
```

Comment:
Can this just be combined? Wouldn't the regex "hey" still match "hey" so if we are going to introduce regex we can just do it for all of them

Not sure if theres a better way to figure out if something is a regex vs just a string to optimize this.
Ex: if i exclude ":p1" and ":p[3-5]" can we determine that the first string is just a string and the second one is a regex

Reply:
I didn't want to break anything already existing. The "String contains" check is against the name, but the regex is against the path.

https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html#getName--

> Returns the name of this project. The project's name is not necessarily unique within a project hierarchy. You should use the [getPath()](https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html#getPath--) method for a unique identifier for the project.